lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -3450,10 +3450,15 @@ ParseResult FIRStmtParser::parsePropAssign() {
   auto rhsType = type_dyn_cast<PropertyType>(rhs.getType());
   if (!lhsType || !rhsType)
     return emitError(loc, "can only propassign property types");
-  if (lhsType != rhsType)
-    return emitError(loc, "cannot propassign non-equivalent type ")
-           << rhsType << " to " << lhsType;
   locationProcessor.setLoc(loc);
+  if (lhsType != rhsType) {
+    // If the lhs is anyref, and the rhs is a ClassType, insert a cast.
+    if (isa<AnyRefType>(lhsType) && isa<ClassType>(rhsType))
```

Comment:
This will reject assignment between `Map<String, Foo>` and `Map<String, AnyRef>`, correct? Do we want to support this kind of subtyping?

Reply:
Yes, that is true. And this is a good question. I haven't thought about subtyping and how we want this to work for container types. This is something to think through--I'm sure there is a sane answer--but I also want to be careful that we maintain the soundness of the property type system. Maybe we can leave this as-is, so it can handle the object reference situation, and then work out how to handle aggregates?

Reply:
Thought about this a little bit more, and I _think_ rejecting the situation you mentioned will be fine for the medium term. The main use case I have in mind is Map<String, AnyRef>, where the port is declared with that type, and we create some objects, then pass them to a map_create op. There will need to be casts for each object (not handled in this PR) that go into the map_create. But I don't foresee needing to assign a Map<String, Foo> to Map<String, AnyRef> in this situation.

Reply:
It's totally reasonable to me to wait until it's actually necessary, thank you for thinking about this!